src/libs/actions/connections/index.ts

```diff
@@ -383,7 +384,8 @@ function getSynchronizationErrorMessage(policy: OnyxEntry<Policy>, connectionNam
     }
 
     const connection = policy?.connections?.[connectionName];
-    if (isSyncInProgress || isEmptyObject(connection?.lastSync) || connection?.lastSync?.isSuccessful) {
+
+    if (isSyncInProgress || isEmptyObject(connection?.lastSync) || connection?.lastSync?.isSuccessful !== false) {
```

Comment:
```suggestion
    if (isSyncInProgress || isEmptyObject(connection?.lastSync) || connection?.lastSync?.isSuccessful !== false || !connection?.lastSync?.errorDate) {
```

I found the root of the problem here. While setting up the Xero connection, we update the connection with some data before the sync has finished. So NewDot receives an update with `connection.lastSync` that has empty/default values since the sync hasn't finished yet. 

So let's make sure both `lastSync.isSuccessful` is `false` AND `lastSync.errorDate` is not empty before showing errors.

Reply:
Cool. Let me test it